backend/compact-connect/lambdas/license-data/tests/function/test_data_model/test_client.py

```diff
@@ -1,20 +1,90 @@
+
 from moto import mock_aws
 from tests.function import TstFunction
 
 
 @mock_aws
 class TestClient(TstFunction):
-    def test_get_ssn(self):
+    def test_get_provider_id(self):
+        from data_model.client import DataClient
+        from data_model.schema.license import LicensePostSchema, LicenseRecordSchema
+        from data_model.schema.privilege import PrivilegePostSchema, PrivilegeRecordSchema
+        from exceptions import CCNotFoundException, CCInternalException
+
+        with open('tests/resources/api/license-post.json', 'r') as f:
+            license_data = LicensePostSchema().load({
+                'compact': 'aslp',
+                'jurisdiction': 'co',
+                **json.load(f)
+            })
+
+        with open('tests/resources/dynamo/license.json', 'r') as f:
+            provider_id = json.load(f)['providerId']
+
+        self._table.put_item(
+            # We'll use the schema/serializer to populate index fields for us
+            Item=LicenseRecordSchema().dump({
+                'providerId': provider_id,
+                'compact': 'aslp',
+                'jurisdiction': 'co',
+                **license_data
+            })
+        )
+
+        with open('tests/resources/api/privilege.json', 'r') as f:
+            privilege = PrivilegePostSchema().loads(f.read())
+
+        self._table.put_item(
+            Item=PrivilegeRecordSchema().dump({
+                'providerId': provider_id,
+                'compact': 'aslp',
+                'jurisdiction': 'co',
+                **privilege
+            })
+        )
+
+        client = DataClient(self.config)
+
+        resp = client.get_provider_id(ssn='123-12-1234')  # pylint: disable=missing-kwoa
+        self.assertEqual(provider_id, resp)
+
+        with self.assertRaises(CCNotFoundException):
+            client.get_provider_id(ssn='321-21-4321')  # pylint: disable=missing-kwoa
+
+        # Associate a second provider with the same ssn - force a data consistency error
+        self._table.put_item(
+            # We'll use the schema/serializer to populate index fields for us
+            Item=LicenseRecordSchema().dump({
+                'providerId': str(uuid4()),
+                'compact': 'aslp',
+                'jurisdiction': 'co',
+                **license_data
+            })
+        )
+        with self.assertRaises(CCInternalException):
+            client.get_provider_id(ssn='123-12-1234')  # pylint: disable=missing-kwoa
+
+    def test_get_provider(self):
         from data_model.client import DataClient
         from data_model.schema.license import LicensePostSchema, LicenseRecordSchema
         from data_model.schema.privilege import PrivilegePostSchema, PrivilegeRecordSchema
 
-        with open('tests/resources/api/license.json', 'r') as f:
-            license_data = LicensePostSchema().loads(f.read())
+        with open('tests/resources/api/license-post.json', 'r') as f:
+            license_data = LicensePostSchema().load({
```

Comment:
Because the test json file represents what should be in the API POST body, where the compact and jurisdiction are provided via alternate means, depending on the point of entry. I wrote the tests to reflect that pattern.

Reply:
This may be a dumb question...but in my brain, I see `license-post.json` and I see `LicensePostSchema`, so I would expect the json file to align with the schema? Doesn't the schema imply that compact and jurisdiction should also be in the API POST body?

(This is just me trying to fully understand.. thanks for bearing with me.)

Reply:
`LicensePostSchema` and `license-post.json` did actually match at first. I ended up moving away from that, eventually. The reason really came down to this: While compact and jurisdiction aren't provided via the 'body' of the two entrypoints of license data (API POST and bulk-upload file), they are always required and provided via a separate avenue (url path parameters for the API, s3 object key for the bulk-upload file). I started by validating them separately from the `LicensePostSchema` until we made the `licenseType` field validation more intelligent in that its allowed values depend on the `compact` value provided.

Now that those two fields are interdependent, it gets much messier to keep one in the schema and the other separate. Ultimately, I decided to refactor the flows to just pull the compact/jurisdiction fields in from the separate sources and combine them, before loading the data through the schema. This has the benefit of reflecting the total information that is required for a new license, even though it may be a bit confusing at first to see fields in the schema that aren't directly provided in the request bodies